src/prefect/task_engine.py

```diff
@@ -448,7 +450,7 @@ def set_state(self, state: State, force: bool = False) -> State:
 
         if new_state.is_final():
             if isinstance(state.data, ResultRecord):
-                result = state.data.result
+                result = state.data.result  # type: ignore[reportUnknownMemberType]
```

Comment:
I think we can get this to work correctly, so let's not ignore it just yet.
```suggestion
                result = state.data.result
```